rotkehlchen/db/filtering.py

```diff
@@ -980,7 +980,7 @@ def make(
                     values=event_identifiers,
                 ),
             )
-        if protocols is not None:
+        if protocols is not None and cls == EvmEventFilterQuery:
             filters.append(
                 DBProtocolsFilter(and_op=True, protocols=protocols),
             )
```

Comment:
Yes, perhaps calling super's make is better. What made me confused about this all filtering is that if we call `HistoryEventFilterQuery`'s make, it would require `EvmEventFilterQuery`'s make to accept the entrie list of all the `HistoryEventFilterQuery` make's arguments to then just pass them along.

-----------

However, choosing from two, I agree with you and think that the current approach is way more ugly. Will change.